src/Stryker.Core/Stryker.Core/Reporters/MarkdownSummaryReporter.cs

```diff
@@ -0,0 +1,133 @@
+using Spectre.Console;
+using Stryker.Core.Mutants;
+using Stryker.Core.Mutators;
+using Stryker.Core.Options;
+using Stryker.Core.ProjectComponents;
+using static FSharp.Compiler.AbstractIL.IL.ILExceptionClause;
+
+namespace Stryker.Core.Reporters
+{
+    /// <summary>
+    /// Markdown result table reporter.
+    /// </summary>
+    public class MarkdownSummaryReporter : IReporter
+    {
+        private readonly StrykerOptions _options;
+        private readonly IAnsiConsole _console;
+        private readonly IFileSystem _fileSystem;
+
+        public MarkdownSummaryReporter(StrykerOptions strykerOptions, IFileSystem fileSystem = null, IAnsiConsole console = null)
+        {
+            _options = strykerOptions;
+            _console = console ?? AnsiConsole.Console;
+            _fileSystem = fileSystem ?? new FileSystem();
+        }
+
+        public void OnStartMutantTestRun(IEnumerable<IReadOnlyMutant> mutantsToBeTested)
+        {
+            // This reporter does not report during the testrun
+        }
+
+        public void OnMutantsCreated(IReadOnlyProjectComponent reportComponent)
+        {
+            // This reporter does not report during the testrun
+        }
+
+        public void OnMutantTested(IReadOnlyMutant result)
+        {
+            // This reporter does not report during the testrun
+        }
+
+        public void OnAllMutantsTested(IReadOnlyProjectComponent reportComponent)
+        {
+            var files = reportComponent.GetAllFiles();
+            if (files.Any())
+            {
+                var filename = _options.ReportFileName + ".md";
+                var reportPath = Path.Combine(_options.ReportPath, filename);
+                var reportUri = "file://" + reportPath.Replace("\\", "/");
+
+                GenerateMarkdownReport(_options, reportPath, files, reportComponent.GetMutationScore());
+
+                _console.WriteLine();
+                _console.MarkupLine("[Green]Your Markdown summary has been generated at:[/]");
+
+                if (_console.Profile.Capabilities.Links)
+                {
+                    // We must print the report path as the link text because on some terminals links might be supported but not actually clickable: https://github.com/spectreconsole/spectre.console/issues/764
+                    _console.MarkupLineInterpolated($"[Green][link={reportUri}]{reportPath}[/][/]");
+                }
+                else
+                {
+                    _console.MarkupLineInterpolated($"[Green]{reportUri}[/]");
+                }
+            }
+        }
+
+        private void GenerateMarkdownReport(StrykerOptions options, string reportPath, IEnumerable<IFileLeaf> files, double mutationScore)
+        {
+            if (!files.Any())
+            {
+                return;
+            }
+
+            var mdSummaryDocument = new MdDocument();
+            mdSummaryDocument.Root.Add(new MdHeading(1, "Mutation Testing Summary"));
```

Comment:
Open to any sort of feedback here. 

- Is this library for Markdown okay with people?
- Is the formatting of the report okay?  (It's not particularly pretty, but does the job at this point)
- Anything I'm missing in this report that I should be adding?

Reply:
The table format looks fine to me. If others would like more information that can always be added later. I will look into the library tomorrow and see if there see any reasons not to use it.

Reply:
I had a look at the library. It seems like the author is still actively maintaining the library even though there hasn't been a release in a while. That's good. I also had a look at alternatives. It seems like Microsoft themselves have moved to https://github.com/xoofx/markdig, however this library seems to be primarily focused on rendering markdown in html. That seems a bit excessive for stryker needs. For that reason I think markdown-generator is fine.